CRM/Core/BAO/CustomField.php

```diff
@@ -976,6 +976,21 @@ public static function deleteField($field) {
     self::deleteRecord(['id' => $field->id]);
   }
 
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param \Civi\Core\Event\PreEvent $event
+   */
+  public static function on_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    // When deleting a custom group, delete all orphaned fields
+    if ($event->entity === 'CustomGroup' && $event->action === 'delete') {
+      $sql = "SELECT id FROM civicrm_custom_field WHERE custom_group_id = " . $event->id;
```

Comment:
This looks pretty scary, like it could be vulnerable to SQL injection. It can be automatically escaped with this:

```
$sql = 'SELECT id FROM civicrm_custom_field WHERE custom_group_id = %1';
$orphanedFields = CRM_Core_DAO::executeQuery($sql, [1 => [$event->id, 'Integer']])->fetchAll();
```

An alternative is to retrieve objects using the higher-level BAO: `CRM_Core_BAO_CustomField::retrieve(['custom_group_id' => $event->id], [])`.

Reply:
Thanks for the review @Sjord. I agree it's a good practice to escape all variables in sql queries, and so I'm happy to make that change just for the sake of "following best practices." However as a practical matter this is not vulnerable to sqli because `$event->id` is guaranteed to be a positive integer; just before the hook is invoked it goes through validation:

https://github.com/civicrm/civicrm-core/blob/fbb81ad15569fb918b186f7e92d7009502bdc89b/CRM/Core/DAO.php#L1115-L1117